core/focus_manager.ts

```diff
@@ -349,8 +356,12 @@ export class FocusManager {
     // happens when calls to focusNode() interleave with asynchronous clean-up
     // operations (which can happen due to ephemeral focus and in other cases).
     // Fall back to a reasonable default since there's no valid node to focus.
+    const focusableNodeElement = focusableNode.getFocusableElement();
+    if (!focusableNodeElement.id || focusableNodeElement.id === 'null') {
+      console.warn('Trying to focus a node that has an invalid ID.');
```

Comment:
Should this clear isUpdatingFocusedNode and return early? It looks like the only time the rest of this method would do anything is if the element is a root node.